sdk/cognitiveservices/textanalytics/Azure.AI.TextAnalytics/src/DetectLanguageResult.cs

```diff
@@ -20,14 +20,15 @@ internal DetectLanguageResult(string id, TextDocumentStatistics statistics, ILis
         internal DetectLanguageResult(string id, string errorMessage)
             : base(id, errorMessage)
         {
+            DetectedLanguages = new ReadOnlyCollection<DetectedLanguage>(new List<DetectedLanguage>());
         }
 
         /// <summary>
         /// </summary>
-        public ReadOnlyCollection<DetectedLanguage> DetectedLanguages { get; }
+        public IReadOnlyCollection<DetectedLanguage> DetectedLanguages { get; }
 
         /// <summary>
         /// </summary>
-        public DetectedLanguage PrimaryLanguage => DetectedLanguages.OrderBy(l => l.Score).First();
+        public DetectedLanguage PrimaryLanguage => DetectedLanguages?.OrderBy(l => l.Score).FirstOrDefault() ?? default;
```

Comment:
Nits:

1. Seems `DetectedLanguages` could never be null given your constructor changes, so checking for null would be unnecessary/wasteful (assuming it's not compiled/JIT'd out).
2. `?? default` is unnecessary either way because, even if `DetectedLanguages` was null, null would simply be returned (same as `default`).